www/makefsdata.js

```diff
@@ -1,41 +1,265 @@
+	['ico', 'image/x-icon'],
+	['class', 'application/octet-stream'],
+	['cls', 'application/octet-stream'],
+	['js', 'application/javascript'],
+	['ram', 'application/javascript'],
+	['css', 'text/css'],
+	['swf', 'application/x-shockwave-flash'],
+	['xml', 'text/xml'],
+	['xsl', 'text/xml'],
+	['pdf', 'application/pdf'],
+	['json', 'application/json'],
+]);
+
+const defaultContentType = 'text/plain';
+
+const shtmlExtensions = new Set(['shtml', 'shtm', 'ssi', 'xml', 'json']);
+
+const skipCompressionExtensions = new Set(['png', 'json']);
+
+const serverHeader = 'GP2040-CE';
 
-const fsdata_filename = path.normalize("/lib/httpd/fsdata.c");
-
-function correctincludes() {
-    fs.readFile(root + fsdata_filename, "utf8", function (error, data) {
-        if (data) {
-            let result = data.replace("#include \"lwip/apps/fs.h\"\r\n#include \"lwip/def.h\"", '#include "fsdata.h"');
-        
-            fs.writeFile(root + fsdata_filename, result, "utf8", function (error) {
-                if (error) {
-                    console.error(error);
-                }
-            });
-        } else {
-            if (error) {
-                console.error(error);
-            }
-        }
-    });
+const payloadAlignment = 4;
+const hexBytesPerLine = 16;
+
+function getFiles(dir) {
+	let results = [];
+	const list = fs.readdirSync(dir, { withFileTypes: true });
+	for (const file of list) {
+		file.path = dir + '/' + file.name;
+		if (file.isDirectory()) {
+			results = results.concat(getFiles(file.path));
+		} else if (file.isFile()) {
+			results.push(file.path);
+		}
+	}
+	return results;
+}
+
+function getLowerCaseFileExtension(file) {
+	const ext = file.split('.').pop();
+	return ext ? ext.toLowerCase() : '';
+}
+
+function fixFilenameForC(filename) {
+	let varName = '';
+	for (let i = 0; i < filename.length; i++) {
+		const c = filename[i];
+		if (c >= 'a' && c <= 'z') {
+			varName += c;
+		} else if (c >= 'A' && c <= 'Z') {
+			varName += c;
+		} else if (c >= '0' && c <= '9') {
+			varName += c;
+		} else if (c === '_') {
+			varName += c;
+		} else {
+			varName += '_';
+		}
+	}
+	return varName;
+}
+
+function CStringLength(str) {
+	return Buffer.from(str, 'utf8').length;
+}
+
+function createHexString(value, prependComment = false) {
+	let column = 0;
+	let hexString = '';
+	if (typeof value === 'string') {
+		const bytes = Buffer.from(value, 'utf8');
+		if (prependComment) {
+			hexString += `/* "${value}" (${bytes.length} bytes) */\n`;
+		}
+		bytes.forEach((byte) => {
+			hexString += '0x' + byte.toString(16).padStart(2, '0') + ',';
+			if (++column >= hexBytesPerLine) {
+				hexString += '\n';
+				column = 0;
+			}
+		});
+	} else if (Buffer.isBuffer(value) || value instanceof Uint8Array) {
+		value.forEach((byte) => {
+			hexString += '0x' + byte.toString(16).padStart(2, '0') + ',';
+			if (++column >= hexBytesPerLine) {
+				hexString += '\n';
+				column = 0;
+			}
+		});
+	} else {
+		assert(false, 'Invalid value type');
```

Comment:
A recommendation would be to not use assert, it's usually for test assertion/test suits.
It would throw some error that might be confusing like: 
```
 actual: false,
 expected: true,
```

instead create an error and throw it:
``` javascript
throw new Error('Invalid value type')
```

Reply:
Changed with the latest commeit.